File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/interfaces/RoomStreamEvent.java
package com.syahputrareno975.cardbattlemodule.interfaces;

import com.syahputrareno975.cardbattlemodule.model.player.PlayerModel;
import com.syahputrareno975.cardbattlemodule.model.playerBattleResult.AllPlayerBattleResultModel;
import com.syahputrareno975.cardbattlemodule.model.room.RoomDataModel;
import com.syahputrareno975.cardbattlemodule.model.roomResult.EndResultModel;

public interface RoomStreamEvent {
    void onConnected(RoomStreamEventController c);
    void onPlayerJoin(PlayerModel p);
    void onPlayerLeft(PlayerModel p);
    void onRoomUpdate(RoomDataModel r);
    void onCountDown(int i , RoomDataModel r);
    void onBattleResult(AllPlayerBattleResultModel r);
    void onResult(EndResultModel r);
    void onDraw(int flag);
    void onGetRoomData(RoomDataModel r);
    void onDisconnected();
    void onError(String s);
    void onException(String e,int flag);
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/interfaces/RoomStreamEvent.kt
package com.syahputrareno975.simplecardbattle.interfaces

import com.syahputrareno975.simplecardbattle.model.player.PlayerModel
import com.syahputrareno975.simplecardbattle.model.playerBattleResult.AllPlayerBattleResultModel
import com.syahputrareno975.simplecardbattle.model.room.RoomDataModel
import com.syahputrareno975.simplecardbattle.model.roomResult.EndResultModel

interface RoomStreamEvent {
    fun onConnected(c : RoomStreamEventController)
    fun onPlayerJoin(p : PlayerModel)
    fun onPlayerLeft(p : PlayerModel)
    fun onRoomUpdate(r : RoomDataModel)
    fun onCountDown(i : Int ,r : RoomDataModel)
    fun onBattleResult(r : AllPlayerBattleResultModel)
    fun onResult(r : EndResultModel)
    fun onDraw(flag : Int)
    fun onGetRoomData(r : RoomDataModel)
    fun onDisconnected()
    fun onError(s : String)
    fun onException(e : String,flag : Int)
}